CMakeLists.txt

```diff
@@ -248,6 +250,17 @@ target_compile_definitions(
   PUBLIC APP_VERSION="${PROJECT_VERSION}${GIT_REV}"
          QT_DISABLE_DEPRECATED_BEFORE=0x050900 QSU_INCLUDE_MOC=1)
 
+# Enable useful compiler warnings (except on third-party code).
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES
+                                           "Clang")
+  target_compile_options(${PROJECT_NAME} PUBLIC -Wall -Wextra -Wpedantic
```

Comment:
Assuming this gets approved at all, what do guys think about treating all warnings as errors from now on?

Reply:
As long as it is only enabled for debug builds, since we don't want packages failing to build because of that.

Reply:
I agree. Enabling these warnings and enforcing them will improve code quality.

Reply:
@bjorn While that's an easy check to add to CMake, our Windows pipeline doesn't run debug builds, so these extra warnings will be essentially useless...

Wouldn't it be desirable to have this enabled for release builds as well, though? Especially considering that all these warnings are now fixed?

In any case, if you think it's going to cause more harm than good, I'm also fine with leaving them as just warnings.

Reply:
Okay, for now I think we can leave this as-is.

Our CI pipeline will generate annotations whenever the compiler gives us warnings, so I think that's good enough and not disruptive...

Thanks!